packages/core/src/RenderPipeline/RenderQueue.ts

```diff
@@ -172,6 +188,8 @@ export class RenderQueue {
         );
 
         rhi.drawPrimitive(primitive, subElement.subPrimitive, program);
+        // Restore the previous stencil state from temp stencil state
+        maskInteractionNotNone && this._copyStencilState(tempStencilState, stencilState);
```

Comment:
Can this logic be written in `StencilState`? It can improve a lot of performance.

Reply:
> Can this logic be written in `StencilState`? It can improve a lot of performance.

Good idea, but i have some misgiving：
1、Is it really appropriate to place such specialized logic inside StencilState? If there are similar cases in the future where dynamic state changes are needed, should we keep adding various switches to State?
2、In most cases, there are only a few masks. Is the performance gain from this part worth it?

Reply:
1. It is not written rigidly in RenderState, but has an elegant replacement logic that does not modify the original RenderState.
![image](https://github.com/user-attachments/assets/312c99ce-e452-4116-8f66-b9114859d2bd)
2. In 2D projects, the frequency of using masks is very high

Reply:
done